components/integrations/bots/add_bot/add_bot.tsx

```diff
@@ -13,178 +13,201 @@ import {General} from 'mattermost-redux/constants';
+            displayName: this.props.bot ? this.props.bot.display_name : '',
+            description: this.props.bot ? this.props.bot.description : '',
             adding: false,
             image: BotDefaultIcon,
-            role: roleOptionMember,
-            postAll: false,
-            postChannels: false,
+            role: UserUtils.isSystemAdmin(this.props.roles || '') ? roleOptionSystemAdmin : roleOptionMember,
+            postAll: this.props.bot ? UserUtils.hasPostAllRole(this.props.roles || '') : false,
+            postChannels: this.props.bot ? UserUtils.hasPostAllPublicRole(this.props.roles || '') : false,
+            orientationStyles: {transform: '', transformOrigin: ''},
+            pictureFile: null,
         };
-
-        if (this.props.bot) {
-            this.state.username = this.props.bot.username;
-            this.state.displayName = this.props.bot.display_name;
-            this.state.description = this.props.bot.description;
-            this.state.role = UserUtils.isSystemAdmin(this.props.roles || '') ? roleOptionSystemAdmin : roleOptionMember;
-            this.state.postAll = UserUtils.hasPostAllRole(this.props.roles || '');
-            this.state.postChannels = UserUtils.hasPostAllPublicRole(this.props.roles || '');
-        }
     }
 
-    updateUsername = (e) => {
+    updateUsername = (e: ChangeEvent<HTMLInputElement>) => {
         this.setState({
             username: e.target.value,
         });
     }
 
-    updateDisplayName = (e) => {
+    updateDisplayName = (e: ChangeEvent<HTMLInputElement>) => {
         this.setState({
             displayName: e.target.value,
         });
     }
 
-    updateDescription = (e) => {
+    updateDescription = (e: ChangeEvent<HTMLInputElement>) => {
         this.setState({
             description: e.target.value,
         });
     }
 
-    updateRole = (e) => {
+    updateRole = (e: ChangeEvent<HTMLSelectElement>) => {
         this.setState({
             role: e.target.value,
         });
     }
 
-    updatePostAll = (e) => {
+    updatePostAll = (e: ChangeEvent<HTMLInputElement>) => {
         this.setState({
             postAll: e.target.checked,
         });
     }
 
-    updatePostChannels = (e) => {
+    updatePostChannels = (e: ChangeEvent<HTMLInputElement>) => {
         this.setState({
             postChannels: e.target.checked,
         });
     }
 
-    updatePicture = (e) => {
+    updatePicture = (e: ChangeEvent<HTMLInputElement>) => {
         if (e.target.files && e.target.files[0]) {
             const pictureFile = e.target.files[0];
             this.previewBlob = URL.createObjectURL(pictureFile);
 
-            var reader = new FileReader();
+            const reader = new FileReader();
             reader.onload = (e2) => {
-                const orientation = FileUtils.getExifOrientation(e2.target.result);
+                const orientation = FileUtils.getExifOrientation(e2.target?.result as ArrayBuffer);
```

Comment:
Casting to `ArrayBuffer` is not the best but I see similar usage here: https://github.com/mattermost/mattermost-webapp/blob/c3a979db979b2f3da8f4bb986dde6b0d43dc11ea/components/picture_selector.tsx#L41

Please let me know if you have any suggestion to improve

Reply:
That should be fine. I suppose we could null check `e2.target.result`, but based on [this](https://developer.mozilla.org/en-US/docs/Web/API/FileReader/result), it should never be null by this point